File: uri/estructuras/tda_racional/1022.cpp
#include <iostream>
using namespace std;

long long mcd(long long a, long long b) {
    if (a < 0) a = -a;
    if (b < 0) b = -b;
    while (b) {
        long long t = b;
        b = a % b;
        a = t;
    }
    return a;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n;
    cin >> n;
    
    while (n--) {
        long long n1, d1, n2, d2;
        char barra, op;
        
        cin >> n1 >> barra >> d1 >> op >> n2 >> barra >> d2;
        
        long long num, den;
        
        if (op == '+') {
            num = n1 * d2 + n2 * d1;
            den = d1 * d2;
        } else if (op == '-') {
            num = n1 * d2 - n2 * d1;
            den = d1 * d2;
        } else if (op == '*') {
            num = n1 * n2;
            den = d1 * d2;
        } else {
            num = n1 * d2;
            den = n2 * d1;
        }
        
        long long g = mcd(num, den);
        long long numSimp = num / g;
        long long denSimp = den / g;
        
        if (denSimp < 0) {
            numSimp = -numSimp;
            denSimp = -denSimp;
        }
        
        cout << num << "/" << den << " = " << numSimp << "/" << denSimp << '\n';
    }
    
    return 0;
}

File: uri/estructuras/tda_racional/1022.rs
use std::io::{self, BufRead, Write, BufWriter};

fn mcd(mut a: i64, mut b: i64) -> i64 {
    if a < 0 { a = -a; }
    if (b < 0) { b = -b; }
    while b != 0 {
        let t = b;
        b = a % b;
        a = t;
    }
    a
}

fn main() {
    let stdin = io::stdin();
    let stdout = io::stdout();
    let mut out = BufWriter::new(stdout.lock());
    let mut lines = stdin.lock().lines();
    
    let n: i32 = lines.next().unwrap().unwrap().trim().parse().unwrap();
    
    for _ in 0..n {
        let line = lines.next().unwrap().unwrap();
        let parts: Vec<&str> = line.split_whitespace().collect();
        
        let n1: i64 = parts[0].parse().unwrap();
        let d1: i64 = parts[2].parse().unwrap();
        let op = parts[3].chars().next().unwrap();
        let n2: i64 = parts[4].parse().unwrap();
        let d2: i64 = parts[6].parse().unwrap();
        
        let (num, den) = match op {
            '+' => (n1 * d2 + n2 * d1, d1 * d2),
            '-' => (n1 * d2 - n2 * d1, d1 * d2),
            '*' => (n1 * n2, d1 * d2),
            '/' => (n1 * d2, n2 * d1),
            _ => (0, 1),
        };
        
        let g = mcd(num, den);
        let mut num_simp = num / g;
        let mut den_simp = den / g;
        
        if den_simp < 0 {
            num_simp = -num_simp;
            den_simp = -den_simp;
        }
        
        writeln!(out, "{}/{} = {}/{}", num, den, num_simp, den_simp).unwrap();
    }
}
